src/Microsoft.AspNetCore.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -35,10 +39,12 @@ public class TagHelperParseTreeRewriter : ISyntaxTreeRewriter
             "wbr"
         };
 
-        private TagHelperDescriptorProvider _provider;
-        private Stack<TagBlockTracker> _trackerStack;
+        private readonly List<KeyValuePair<string, string>> _htmlAttributeTracker;
+        private readonly StringBuilder _attributeValueBuilder;
+        private readonly TagHelperDescriptorProvider _provider;
+        private readonly Stack<TagBlockTracker> _trackerStack;
+        private readonly Stack<BlockBuilder> _blockStack;
```

Comment:
All of these collections or collection wrappers (especially the `StringBuilder`) can grow unbounded. Need to have a maximum size for each and `null` out the field when one grows too large. Use an extreme size but they must have limits.